Add notes for running Grafana locally and examples of Dashboards-as-code

stephen-soltesz opened the pull request:
This change adds a "local" configuration example for running Grafana for testing and loading dashboards generated by "grafanalib" from weaveworks. https://www.weave.works/blog/grafana-dashboards-as-code/

This is a precursor to applying this configuration to our production configs.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/prometheus-support/139)
<!-- Reviewable:end -->

gfr10598 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

This looks ok to me, though a bit circuitous.  Do you think it is likely to get simpler over time?

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[config/local/grafana/README.md, line 35 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/139#-L-IpqGsRqWXfl2J77sO:-L-IpqGsRqWXfl2J77sP:b-cgqqlo) ([raw file](https://github.com/m-lab/prometheus-support/blob/5f56752ec66f52e4b4ed0247d876a0ac52983cba/config/local/grafana/README.md#L35)):*
> ```Markdown
> 
> ```
> docker run -it -p 3000:3000 -v /var/lib/grafana --name grafana-storage \
> ```

Since the rest is so detailed, maybe mention where grafana-storage is created?

---

*[config/local/grafana/README.md, line 64 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/139#-L-IoL-U0sKRKbYZyL8I:-L-IoL-U0sKRKbYZyL8J:b-2a7f9h) ([raw file](https://github.com/m-lab/prometheus-support/blob/5f56752ec66f52e4b4ed0247d876a0ac52983cba/config/local/grafana/README.md#L64)):*
> ```Markdown
> everything. Ask me how I learned this. (╯°□°）╯︵┻━┻
> 
> Visit: http://localhost:3000/
> ```

How does one create an account/pw?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/139#-:-L-Iogwu9VrfIjQlo2uj:b793y75)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
I'm not sure what you mean by circuitous. This is a configuration for local testing. It's what I used to demonstrate that the configuration we want is plausible. And, like the other 'local' configs I wanted to document the steps I used so others could run a local setup easily to experiment on their own as well.

Ultimately, I imagine adding a dashboard to our staging and production instances being as simple as changing our other configs or k8s settings are today. That is, we'll create a PR for a dashboard change, the PR is approved, merged, and auto deployed.

How does this sound?

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[config/local/grafana/README.md, line 35 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/139#-L-IpqGsRqWXfl2J77sO:-L-JMwn_RCNIXVj7d-G1:b-a0qcwc) ([raw file](https://github.com/m-lab/prometheus-support/blob/8aa37028e323931817ad1935100ebe7772ec1cc9/config/local/grafana/README.md#L35)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Since the rest is so detailed, maybe mention where grafana-storage is created?
</blockquote></details>

Added a short note.

---

*[config/local/grafana/README.md, line 64 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/139#-L-IoL-U0sKRKbYZyL8I:-L-JMENdVpzhlr4wGsbE:b-bu4qvy) ([raw file](https://github.com/m-lab/prometheus-support/blob/8aa37028e323931817ad1935100ebe7772ec1cc9/config/local/grafana/README.md#L64)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

How does one create an account/pw?
</blockquote></details>

Ah, the intent is to just use the admin account. But, it's not evident what the admin user is. I've added a note about the default username/ password.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/139#-:-L-JN-MrB2jFssDWwNir:bdgjrtd)*
<!-- Sent from Reviewable.io -->
